cli/src/cmd/cast/run.rs

```diff
@@ -103,7 +103,7 @@ impl RunArgs {
         // configures a bare version of the evm executor: no cheatcode inspector is enabled,
         // tracing will be enabled only for the targeted transaction
         let builder =
-            ExecutorBuilder::default().with_config(env).with_spec(evm_spec(&config.evm_version));
+            ExecutorBuilder::default().with_config(env).with_spec(SpecId::SHANGHAI);
```

Comment:
This is not doing what's intended—it's expected for the debugger to trip up when it sees a `PUSH0` opcode if the evm version is not set to shanghai. 

@Code0x2 doesn't the debugger work with `--evm-version shanghai` on the cli or `evm_version=shanghai` on your `foundry.toml`?

Reply:
> it's expected for the debugger to trip up when it sees a PUSH0 opcode if the evm version is not set to shanghai.

Hm, I'm not convinced, this is kind of tricky: https://github.com/foundry-rs/foundry/pull/5345#discussion_r1258413088

> doesn't the debugger work with `--evm-version shanghai` on the cli or `evm_version=shanghai` on your `foundry.toml`? I believe we do not need this change at all—you just need to set the appropiate evm version.

This could also be a good solution: in my example there you'd error on push0, explain to the user why, and suggest re-running with the above flag

Reply:
@Evalir Im using the debugger from `cast run -d -r <rpc> <txhash>`, and there is no option there to set evm version

Maybe the better way is to allow setting evm version from cast cli